packages/core/src/shared/extensionUtilities.ts

```diff
@@ -170,12 +170,31 @@ export function isCloud9(flavor: 'classic' | 'codecatalyst' | 'any' = 'any'): bo
     return (flavor === 'classic' && !codecat) || (flavor === 'codecatalyst' && codecat)
 }
 
+/**
+ * Checks if the current environment has SageMaker-specific environment variables
+ * @returns true if SageMaker environment variables are detected
+ */
+function hasSageMakerEnvVars(): boolean {
+    return (
+        process.env.SAGEMAKER_APP_TYPE !== undefined ||
```

Comment:
non blocking: This need to update as SAGEMAKER changes their env var in their docket img..